components/Posts/BlogMenuCat/index.tsx

```diff
@@ -7,7 +7,24 @@ interface Props {
 
 export const BlogMenuCat: FC<Props> = ( { categories, onSearch } ) =>
 {
-	const [activeCategory, setActiveCategory] = useState<string | undefined>("all");
+	const [activeCategory, setActiveCategory] = useState<string>("");
```

Comment:
we don't want to repeat the code, please create a custom hook for this. 

and those hooks you can place in 
lib/hooks/hookname.ts